src/dynd/func/fill.cpp

```diff
@@ -0,0 +1,109 @@
+//
+// Copyright (C) 2011-15 DyND Developers
+// BSD 2-Clause License, see LICENSE.txt
+//
+
+#include <dynd/func/fill.hpp>
+#include <dynd/func/elwise.hpp>
+#include <dynd/func/multidispatch.hpp>
+#include <dynd/func/call.hpp>
+#include <dynd/kernels/zeros_kernel.hpp>
+#include <dynd/kernels/ones_kernel.hpp>
+
+using namespace std;
+using namespace dynd;
+
+
+DYND_API nd::callable nd::zeros::children[DYND_TYPE_ID_MAX + 1];
+
+DYND_API nd::callable nd::zeros::make()
+{
+    typedef type_id_sequence<
+        bool_type_id,
+        int8_type_id,
+        int16_type_id,
+        int32_type_id,
+        int64_type_id,
+        int128_type_id,
+        uint8_type_id,
+        uint16_type_id,
+        uint32_type_id,
+        uint64_type_id,
+        uint128_type_id,
+        float16_type_id,
+        float32_type_id,
+        float64_type_id,
+        float128_type_id,
+        complex_float32_type_id,
+        complex_float64_type_id
+    > type_ids;
```

Comment:
We should be able to just use some of our predefined typedefs here. [aritmetic_type_ids](https://github.com/libdynd/libdynd/blob/master/include/dynd/types/type_id.hpp#L207) is the same as what you define here.